var/spack/repos/builtin/packages/pism/package.py

```diff
@@ -78,22 +79,26 @@ class Pism(CMakePackage):
     depends_on('petsc')
     depends_on('udunits')
     depends_on('proj@:4', when='@:1.1')
-    depends_on('proj@6:', when='@1.2:')
+    depends_on('proj@5:', when='@1.2:')   # Can use PROJ.4 or PROJ.5+ API
     depends_on('everytrace', when='+everytrace')
 
-    extends('python', when='+python')
-    depends_on('python@2.7:2.8,3.3:', when='@1.1: +python')
-    depends_on('python@2.7:2.8', when='@:1.0 +python')
-    depends_on('py-matplotlib', when='+python')
-    depends_on('py-numpy', when='+python')
+    extends('python', type=('build', 'run'), when='+python')
```

Comment:
The `type` kwarg for `extends` currently doesn't do anything, but it will work the way you expect once #20045 is merged.